### ISSUE TITLE
Report of autocomplete suggestion from history lingering after nonmatching input

### ISSUE THREAD

**Issue by ekager** [mozilla-mobile/android-components#3611]
Following is an example but not the exact URL scenario. @dholbert says it seems to be happening infrequently with history URLs.

"I started typing "yahoo", and y autocompleted to yelp.com, and then I typed "ah" and I was briefly looking at "yah[elp.com]" but then after ~1 second of looking at that, it fixed itself"

STR:
When opening Fenix (usually from fresh launch) 
1. Open search
2. Type "a" ("[mazon.com]" suggested)
3. Type something that doesn't match the suggestion, ie "aa"

Expected: 
Suggestion updates

Actual: 
"[mazon.com]" remains as the autocomplete suggestion like "aaa[mazon.com]". Needs to be backspaced to delete it

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-2955)

**Comment by dholbert** [comment]
Here's a screencast of me triggering the bug (very briefly) when starting to type "yahoo":
https://www.dropbox.com/s/84w8tj3jnfouii7/fenix-autocomplete.webm?dl=0

The bug reproduces just before the ~0:02 timestamp in that video. If you watch the video through slowly (e.g. clicking play/pause, play/pause, to look at intermediate renderings), you can see these various states in the URL bar (where I'm using [bracketed text] to indicate an autocomplete suggestion):
 (a) y
 (b) y[elp.com]
 (c) ya[elp.com]
 (d) yah[elp.com]
 (e) yah[oo.com]

States (c) and (d) are bogus and should never actually show up to the user. If the user were to press enter at the wrong time, it looks like they might accidentally end up at a bogus "franken-site" that we did not intend to suggest.  (I'm not sure if that's actually what would happen, but that's how it looks now.)

In this screencast, I'm only in the broken state for a fraction of a second, but I've experienced this bug where the bogus autocomplete is around for at least a second. (Enough time for me to notice, get irritated, and hit backspace to get rid of it.)

**Comment by dholbert** [comment]
Here's another screencast I just captured where the issue is around for a bit longer:
https://www.dropbox.com/s/prhbk0bf8l5j0ty/fenix-autocomplete2.webm?dl=0

I'm starting to type "imdb.com", and I get an initial suggestion for "isthereagiantsgametoday.com", and this autocompletion sticks around for nearly a full second after I've typed the "m".

So I'm left looking at "im[hereagiantsgametoday.com]" (a clearly-bogus suggestion) for an appreciable amount of time.

**Comment by ekager** [comment]
Also possibly related to Fenix not using the AwesomeBarFeature https://github.com/mozilla-mobile/fenix/issues/2406#issuecomment-497089188 but we should investigate more. Thanks for the video capture and info! :)

**Comment by dholbert** [comment]
VERSION INFO:
- I'm using Firefox Preview 1.0.1926 (Build #11790618) built Friday 6/28 @ 6:17 AM.
- Environment: Pixel 3, running Android 10 "Q" beta
- I'm been reproducing by:
 (1) kill fenix
 (2) open fenix
 (3) immediately tap URL bar and start typing a domain name (and be sure that the second character disagrees with the autocomplete suggestion for the first character).

With these steps, I've been able to reproduce the bug within a few minutes of continuous attempts. (Most of the times, the autocomplete appears & changes instantly when I type the second character, which is great. And some of the times, I don't see any autocompletion until I've typed the second character, which is also fine.)
